src/axom/primal/tests/primal_clip.cpp

```diff
@@ -329,6 +329,12 @@ void unit_check_poly_clip()
   //      |
   // In addition, vertices 0 and 3 should be marked as clipped.
 
+  const int current_allocator = axom::getDefaultAllocatorID();
```

Comment:
Similarly here: 
Do we have a better pattern for allocating/deallocating moving a single object between host and device, e.g. using a `StackArray` ?

(Not required for this PR, but if we have a cleaner way of doing this, we should prefer to use it in the unit tests)

Reply:
> Do we have a better pattern for allocating/deallocating moving a single object between host and device, e.g. using a StackArray ?

Not that I am aware of. The best I have found is either using `axom::allocate` & `axom::copy`, or using `axom::Array` constructors to move between host and device.